website/common/script/libs/achievements.js

```diff
@@ -0,0 +1,306 @@
+      contribTitle = i18n.t('elite', language);
+    } else if (lvl < 7) {
+      contribTitle = i18n.t('champion', language);
+    } else if (lvl < 8) {
+      contribTitle = i18n.t('legendary', language);
+    } else if (lvl < 9) {
+      contribTitle = i18n.t('guardian', language);
+    } else {
+      contribTitle = i18n.t('heroic', language);
+    }
+
+    return `${contribTitle} ${contrib.text}`;
+  }
+}
+
+function _add (result, data) {
+  result[data.key] = {
+    title: data.title,
+    text: data.text,
+    icon: data.icon,
+    earned: data.earned,
+    value: data.value,
+    index: index++,
+    optionalCount: data.optionalCount,
+  };
+}
+
+function _addSimpleWithCustomPath (result, user, data) {
+  let value = get(user, data.path);
+  let thisContent = achievsContent[data.key];
+
+  _add(result, {
+    title: i18n.t(thisContent.titleKey, {key: value}, data.language),
+    text: i18n.t(thisContent.textKey, data.language),
+    icon: thisContent.icon,
+    key: data.key,
+    value,
+    earned: Boolean(value),
+  });
+}
+
+function _addQuest (result, user, data) {
+  data.key = `${data.path}Quest`;
+  data.path = `achievements.quests.${data.path}`;
+  _addSimpleWithCustomPath(result, user, data);
+}
+
+function _addSimple (result, user, data) {
+  let value = user.achievements[data.path];
+
+  let key = data.key || data.path;
+  let thisContent = achievsContent[key];
+
+  _add(result, {
+    title: i18n.t(thisContent.titleKey, data.language),
+    text: i18n.t(thisContent.textKey, data.language),
+    icon: thisContent.icon,
+    key,
+    value,
+    earned: Boolean(value),
+  });
+}
+
+function _addSimpleWithMasterCount (result, user, data) {
+  let language = data.language;
+  let value = user.achievements[`${data.path}Count`] || 0;
+
+  let thisContent = achievsContent[data.path];
+
+  let text = i18n.t(thisContent.textKey, language);
+  if (value > 0) {
+    text += i18n.t(thisContent.text2Key, {count: value}, language);
+  }
+
+  _add(result, {
+    title: i18n.t(thisContent.titleKey, language),
+    text,
+    icon: thisContent.icon,
+    key: data.path,
+    value,
+    optionalCount: value,
+    earned: Boolean(value),
+  });
+}
+
+function _addSimpleWithCount (result, user, data) {
+  let value = user.achievements[data.path] || 0;
+
+  let key = data.key || data.path;
+  let thisContent = achievsContent[key];
+
+  _add(result, {
+    title: i18n.t(thisContent.titleKey, data.language),
+    text: i18n.t(thisContent.textKey, {count: value}, data.language),
+    icon: thisContent.icon,
+    key,
+    value,
+    optionalCount: value,
+    earned: Boolean(value),
+  });
+}
+
+function _addPlural (result, user, data) {
+  let value = user.achievements[data.path] || 0;
+
+  let key = data.key || data.path;
+  let thisContent = achievsContent[key];
+
+  let titleKey;
+  let textKey;
+  // If value === 0, use singular versions of strings.
+  // If value !== 0, use plural versions of strings.
+  if (value === 0) {
```

Comment:
@negue I'm looking at this again, and I think this should actually be `=== 1`.

That way, both zero count and multiple count use the plural versions of strings, and ONLY single count uses the singular versions:
- 0 thing`s`
- 1 thing`_`
- 2+ thing`s`